webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/reports/StatusReports.scala

```diff
@@ -357,14 +356,27 @@ final case class BlockStatusReport(
       // simple weighted compliance, as usual
       case WeightedReport => ComplianceLevel.sum(subComponents.map(_.compliance))
       // worst case bubble up, and its weight can be either 1 or the sum of sub-component weight
-      case worst: WorstReportReportingLogic =>
+      case worst: WorstReportWeightedReportingLogic =>
         val worstReport = ReportType.getWorseType(subComponents.map(_.status))
         val allReports  = getValues(_ => true).flatMap(_.messages.map(_ => worstReport))
         val kept        = worst match {
           case WorstReportWeightedOne => allReports.take(1)
           case WorstReportWeightedSum => allReports
         }
         ComplianceLevel.compute(kept)
+      case WorstReportByPercent =>
+        val allReports = subComponents.map {
+          case b: BlockStatusReport => {
+            // block compliance will be turned into a percent, and the worst will be taken
+            b.getValues(_ => true).flatMap(_.messages.map(m => m.reportType))
+          }
+          case o: ValueStatusReport => {
+            // value has 1 count
+            List(ReportType.getWorseType(o.getValues(_ => true).flatMap(_.messages.map(_ => o.status))))
+          }
+        }
+        val worst      = allReports.minBy(ComplianceLevel.compute(_).computePercent().compliance)
+        ComplianceLevel.compute(worst)
```

Comment:
I think it's the correct (only one that makes sense) thing to do. But we need to document it much better (internally, but also here)

At least a comment on each object déclaration, perhaps with some ASCII art of an example so that the next dev get the intent intuition (and then can check it with the unit tests)